### PR TITLE
Add github workflow to store slugs to translate

### LINKED ISSUE #531: [Localization Workflow] Create a GitHub action that checks for MDX files with `translate` field in frontmatter
## Description 
We need to create a GitHub action that checks MDX files in our content folder for the `translate` field whenever a new PR is merged. It will then save all the slugs to an artifact. 

## AC
* [ ] Job triggered by each PR into `develop` 
* [ ] Saves the slugs for newly updated content with `translate` frontmatter to the artifact

### PR DESCRIPTION
**Pull request by roadlittledawn**
## Description
Adds a workflow that runs on the `pull_request` event for PRs that contain `.mdx` files. It will check for any pages that has `translate` in the frontmatter and saves them (by locale) to a queue saved in AWS DynamoDB.

## Related issues / PRs
* Closes #531

## To Do
* [x] Combine what's in the queue with what's being added
* [x] Update this PR to target `main` before merging (see `TODO` comments in the workflow file)

### CODE REVIEW

File: scripts/get-translated-files.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      if (data.translate) {
+        data.translate.forEach((locale) => {
+          if (!accum[locale]) {
+            accum[locale] = [];
+          }
+          accum[locale].push(curr);
+        });
+      }
+      return accum;
+    }, {});
```
Comment (zstix) [inline scripts/get-translated-files.js:33]: If you wanted to do this with pure functions:

```js
.reduce((locales, filename) => {
  const file = fs.readFileSync(path.join(process.cwd(), filename));
  const { data } = frontmatter(file);

  return data.translate && data.translate.length
    ? data.translate.reduce((acc, locale) => ({
        ...acc,
        [locale]: [...(acc[locale] || []), filename]
      }), locales)
    : locales;
}, {});
```

File: .github/workflows/get-slugs-to-translate.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name: Add Slugs to Translation Queue
+
+on:
+  pull_request:
+    branches:
+      - campfire/get-translated-slugs # TODO: update to `main`
```
Comment (zstix) [inline .github/workflows/get-slugs-to-translate.yml:6]:
```suggestion
      - main
```

**Review by jerelmiller — APPROVED** [review]
Super :dop:

**Comment by caylahamann** [comment]
this has already been merged in, but this is super cool 😄

**Pull request merged**